src/Time/TimeSteppers/RungeKutta4.cpp

```diff
@@ -0,0 +1,68 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Time/TimeSteppers/RungeKutta4.hpp"
+
+#include <cmath>
+#include <limits>
+
+#include "ErrorHandling/Assert.hpp"
+#include "Time/TimeStepId.hpp"
+
+namespace TimeSteppers {
+
+uint64_t RungeKutta4::number_of_substeps() const noexcept { return 4; }
+
+size_t RungeKutta4::number_of_past_steps() const noexcept { return 0; }
+
+// The growth function for RK4 is (e.g. page 60 of
+// http://www.staff.science.uu.nl/~frank011/Classes/numwisk/ch10.pdf
+//
+//   g = 1 + mu + mu^2 / 2 + mu^3 / 6 + mu^4 / 24,
+//
+// where mu = lambda * dt. The equation dy/dt = -lambda * y evolves
+// stably if |g| < 1. For lambda=-2, chosen so the stable_step() for
+// RK1 (i.e. forward Euler) would be 1, RK4 has a stable step
+// determined by inserting mu->-2 dt into the above equation. Finding the
+// solutions with a numerical root find yields a stable step of about 1.39265.
+double RungeKutta4::stable_step() const noexcept { return 1.3926467817026411; }
+
+TimeStepId RungeKutta4::next_time_id(const TimeStepId& current_id,
+                                     const TimeDelta& time_step) const
+    noexcept {
+  switch (current_id.substep()) {
+    case 0:
+      ASSERT(current_id.substep_time() == current_id.step_time(),
+             "Wrong substep time");
```

Comment:
Could you print the given time and the expected time? I've personally found errors that don't provide that sort of info rather frustrating :)